eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/Abstraction/HttpPipelineProvider.cs

```diff
@@ -40,10 +43,40 @@ public override ValueExpression PerRetryPolicy(params ValueExpression[] argument
 
         public override ClientPipelineApi ToExpression() => this;
 
-        public override MethodBodyStatement Send(HttpMessageApi message, HttpRequestOptionsApi options)
-            => Original.Invoke(nameof(HttpPipeline.Send), [message, Default]).Terminate();
+        public override MethodBodyStatement[] ProcessMessage(HttpMessageApi message, HttpRequestOptionsApi options)
+        {
+            var userCacellationToken = new ParameterProvider("userCancellationToken", $"", new CSharpType(typeof(CancellationToken)));
+            var statusOption = new ParameterProvider("statusOption", $"", new CSharpType(typeof(ErrorOptions)));
+            return new MethodBodyStatement[]
+            {
+                new VariableTupleExpression(false, userCacellationToken, statusOption).Assign(options.Invoke("Parse")).Terminate(),
+                Original.Invoke(nameof(HttpPipeline.Send), [message, userCacellationToken]).Terminate(),
+                MethodBodyStatement.EmptyLine,
+                new IfStatement(message.Response().IsError().And(new BinaryOperatorExpression("&", options.NullConditional().Property("ErrorOptions"), options.NoThrow()).NotEqual(options.NoThrow())))
+                {
+                    Throw(New.Instance(AzureClientPlugin.Instance.TypeFactory.ClientResponseApi.ClientResponseExceptionType, message.Response()))
+                },
+                MethodBodyStatement.EmptyLine,
+                Return(message.Response())
+            };
+        }
 
-        public override MethodBodyStatement SendAsync(HttpMessageApi message, HttpRequestOptionsApi options)
-            => Original.Invoke(nameof(HttpPipeline.SendAsync), [message, Default], true).Terminate();
+        public override MethodBodyStatement[] ProcessMessageAsync(HttpMessageApi message, HttpRequestOptionsApi options)
```

Comment:
Can we try to avoid duplicating this code between the sync/async methods? Looks like there is just a few things that will need to differ.

Reply:
consolidated